Objects/exceptions.c

```diff
@@ -1881,12 +1907,23 @@ static PyMethodDef ImportError_methods[] = {
     {NULL}
 };
 
-ComplexExtendsException(PyExc_Exception, ImportError,
-                        ImportError, 0 /* new */,
-                        ImportError_methods, ImportError_members,
-                        0 /* getset */, ImportError_str,
-                        "Import can't find module, or can't find name in "
-                        "module.");
+static PyTypeObject _PyExc_ImportError = {
+    PyVarObject_HEAD_INIT(NULL, 0)
+    "ImportError",
+    sizeof(PyImportErrorObject), 0,
+    (destructor)ImportError_dealloc, 0, 0, 0, 0,
```

Comment:
Let's avoid using function casts that may suppress UBSans in the future. IOW, you don't need the `destructor`, `traverseproc`, `inquiry` and `iniproc` casts as the functions should have the correct signatures now.